malasakit-django/pcari/models.py

```diff
@@ -267,14 +278,70 @@ class QuantitativeQuestion(Question):
     Attributes:
         left_text: The text that is rendered on the left end of the slider.
         right_text: The text that is rendered on the right end of the slider.
+        min_val: The smallest possible value for any response.
+        max_val: The largest possible value for any response.
+        input_type: `'range'` or `'number'`
+            range: The question displays as a slider.
+            number: The question displays as a number-specific text field.
     """
+    INPUT_TYPES = (
+        ('range', 'range'),
+        ('number', 'number')
+    )
+
     left_text = models.TextField(blank=True)
     right_text = models.TextField(blank=True)
+    min_val = models.PositiveSmallIntegerField(default=0, null=True, blank=True)
+    max_val = models.PositiveSmallIntegerField(default=9, null=True, blank=True)
+    input_type = models.CharField(max_length=16, choices=INPUT_TYPES,
+                                  default='range', blank=True)
 
     def __unicode__(self):
         return 'QuantitativeQuestion {0}: "{1}"'.format(self.id, self.prompt)
 
 
+@validate_has_input_type
+class OptionQuestion(Question):
+    """
+    An `OptionQuestion` is a `Question` that asks the user to select one option
+    from a pre-defined set of choices.
+
+    Attributes:
+        options: The serialized list of allowed options.
+    """
+    INPUT_TYPES = (
+        ('dropdown', 'dropdown'),
+        ('bubble', 'bubble')
```

Comment:
If we're going to use `range` and `number`, the standard HTML names for those input elements, we should be consistent and use `select` and `radio`.